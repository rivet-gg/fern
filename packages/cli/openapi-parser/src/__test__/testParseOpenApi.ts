import { AbsoluteFilePath, join, RelativeFilePath } from "@fern-api/fs-utils";
import { CONSOLE_LOGGER } from "@fern-api/logger";
import { serialization } from "@fern-api/openapi-ir-sdk";
import { createMockTaskContext } from "@fern-api/task-context";
import { parse, SpecImportSettings } from "../parse";

const FIXTURES_PATH = join(AbsoluteFilePath.of(__dirname), RelativeFilePath.of("fixtures"));

// eslint-disable-next-line jest/no-export
export function testParseOpenAPI(
    fixtureName: string,
    openApiFilename: string,
    asyncApiFilename?: string,
    settings?: SpecImportSettings
): void {
    // eslint-disable-next-line jest/valid-title
    describe(fixtureName, () => {
        it("parse open api", async () => {
            const absolutePathToOpenAPI = join(
                FIXTURES_PATH,
                RelativeFilePath.of(fixtureName),
                RelativeFilePath.of(openApiFilename)
            );
            const absolutePathToAsyncAPI =
                asyncApiFilename != null
                    ? join(FIXTURES_PATH, RelativeFilePath.of(fixtureName), RelativeFilePath.of(asyncApiFilename))
                    : undefined;
            const specs = [];
            if (absolutePathToOpenAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToOpenAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }
            if (absolutePathToAsyncAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToAsyncAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }

            const openApiIr = await parse({
                specs,
                taskContext: createMockTaskContext({ logger: CONSOLE_LOGGER })
            });
            const openApiIrJson = await serialization.OpenApiIntermediateRepresentation.jsonOrThrow(openApiIr, {
                skipValidation: true
            });
            expect(openApiIrJson).toMatchSnapshot();
        });
    });
}
