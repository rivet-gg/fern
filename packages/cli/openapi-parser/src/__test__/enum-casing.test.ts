import { testParseOpenAPI } from "./testParseOpenApi";

describe("examples", () => {
    testParseOpenAPI("enum-casing", "openapi.yml");
});
