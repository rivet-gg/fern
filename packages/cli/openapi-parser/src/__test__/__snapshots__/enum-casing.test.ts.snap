// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples enum-casing parse open api 1`] = `
{
  "apiVersion": null,
  "basePath": null,
  "channel": [],
  "description": null,
  "endpoints": [
    {
      "audiences": [],
      "authed": false,
      "availability": null,
      "description": null,
      "errors": {},
      "examples": [
        {
          "codeSamples": [],
          "description": null,
          "headers": [],
          "name": null,
          "pathParameters": [],
          "queryParameters": [],
          "request": null,
          "response": {
            "properties": {
              "custom_fields": {
                "type": "map",
                "value": [
                  {
                    "key": {
                      "type": "string",
                      "value": "key",
                    },
                    "value": {
                      "type": "primitive",
                      "value": {
                        "type": "string",
                        "value": "value",
                      },
                    },
                  },
                ],
              },
              "message": {
                "type": "primitive",
                "value": {
                  "type": "string",
                  "value": "message",
                },
              },
              "status": {
                "type": "enum",
                "value": "success",
              },
            },
            "type": "object",
          },
          "type": "full",
        },
      ],
      "generatedRequestName": "GetExampleRequest",
      "headers": [],
      "idempotent": null,
      "internal": null,
      "method": "GET",
      "operationId": null,
      "pagination": null,
      "path": "/example",
      "pathParameters": [],
      "queryParameters": [],
      "request": null,
      "requestNameOverride": null,
      "response": {
        "description": "Successful response",
        "responseProperty": null,
        "schema": {
          "availability": null,
          "description": null,
          "generatedName": "GetExampleResponse",
          "groupName": null,
          "nameOverride": null,
          "schema": "ExampleResponse",
          "type": "reference",
        },
        "type": "json",
      },
      "sdkName": null,
      "server": [],
      "summary": "Get Example",
      "tags": [],
    },
  ],
  "globalHeaders": [],
  "groups": {},
  "hasEndpointsMarkedInternal": false,
  "idempotencyHeaders": [],
  "nonRequestReferencedSchemas": [
    "ExampleResponse",
  ],
  "schemas": {
    "ExampleResponse": {
      "additionalProperties": false,
      "allOf": [],
      "allOfPropertyConflicts": [],
      "availability": null,
      "description": null,
      "generatedName": "ExampleResponse",
      "groupName": null,
      "nameOverride": null,
      "properties": [
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "exampleResponseStatus",
          "key": "status",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "exampleResponseStatus",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "default": null,
              "description": null,
              "generatedName": "ExampleResponseStatus",
              "groupName": null,
              "nameOverride": null,
              "type": "enum",
              "values": [
                {
                  "availability": null,
                  "casing": {
                    "camel": null,
                    "pascal": null,
                    "screamingSnake": null,
                    "snake": null,
                  },
                  "description": null,
                  "generatedName": "success",
                  "nameOverride": null,
                  "value": "success",
                },
                {
                  "availability": null,
                  "casing": {
                    "camel": null,
                    "pascal": null,
                    "screamingSnake": null,
                    "snake": null,
                  },
                  "description": null,
                  "generatedName": "failure",
                  "nameOverride": null,
                  "value": "failure",
                },
              ],
            },
          },
        },
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "exampleResponseMessage",
          "key": "message",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "exampleResponseMessage",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "ExampleResponseMessage",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "format": null,
                "maxLength": null,
                "minLength": null,
                "pattern": null,
                "type": "string",
              },
              "type": "primitive",
            },
          },
        },
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "exampleResponseCustomFields",
          "key": "custom_fields",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "exampleResponseCustomFields",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "ExampleResponseCustomFields",
              "groupName": null,
              "key": {
                "availability": null,
                "description": null,
                "generatedName": "ExampleResponseCustomFieldsKey",
                "groupName": null,
                "nameOverride": null,
                "schema": {
                  "default": null,
                  "format": null,
                  "maxLength": null,
                  "minLength": null,
                  "pattern": null,
                  "type": "string",
                },
              },
              "nameOverride": null,
              "type": "map",
              "value": {
                "availability": null,
                "description": null,
                "generatedName": "ExampleResponseCustomFieldsValue",
                "groupName": null,
                "nameOverride": null,
                "schema": {
                  "default": null,
                  "format": null,
                  "maxLength": null,
                  "minLength": null,
                  "pattern": null,
                  "type": "string",
                },
                "type": "primitive",
              },
            },
          },
        },
      ],
      "type": "object",
    },
    "GrantTypeEnum": {
      "availability": null,
      "default": null,
      "description": null,
      "generatedName": "GrantTypeEnum",
      "groupName": null,
      "nameOverride": null,
      "type": "enum",
      "values": [
        {
          "availability": null,
          "casing": {
            "camel": null,
            "pascal": null,
            "screamingSnake": null,
            "snake": null,
          },
          "description": null,
          "generatedName": "UrnDevrevParamsOauthGrantTypeTokenIssue",
          "nameOverride": null,
          "value": "urn:devrev:params:oauth:grant-type:token-issue",
        },
        {
          "availability": null,
          "casing": {
            "camel": null,
            "pascal": null,
            "screamingSnake": null,
            "snake": null,
          },
          "description": null,
          "generatedName": "UrnIetfParamsOauthGrantTypeTokenExchange",
          "nameOverride": null,
          "value": "urn:ietf:params:oauth:grant-type:token-exchange",
        },
      ],
    },
  },
  "securitySchemes": {},
  "servers": [],
  "tags": {
    "orderedTagIds": null,
    "tagsById": {},
  },
  "title": "Example API",
  "variables": {},
  "webhooks": [],
}
`;
